Source/SwiftLintFramework/Rules/DynamicInlineRule.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  DynamicInlineRule.swift
+//  SwiftLint
+//
+//  Created by Daniel Duan on 12/08/16.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct DynamicInlineRule: ASTRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.error)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "dynamic_inline",
+        name: "Dynamic Inline",
+        description: "avoid using 'dynamic' and '@inline(__always)' together.",
+        nonTriggeringExamples: [
+            "class C {\ndynamic func f() {}}",
+            "class C {\n@inline(__always) func f() {}}",
+            "class C {\n@inline(never) dynamic func f() {}}"
+        ],
+        triggeringExamples: [
+            "class C {\n@inline(__always) dynamic func f() {}\n}",
+            "class C {\n@inline(__always) public dynamic func f() {}\n}",
+            "class C {\n@inline(__always) dynamic internal func f() {}\n}",
+            "class C {\n@inline(__always)\ndynamic func f() {}\n}",
+            "class C {\n@inline(__always)\ndynamic\nfunc f() {}\n}"
+        ]
+    )
+
+    public func validateFile(_ file: File, kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        // Look for functions with both "inline" and "dynamic". For each of these, we can get offset
+        // of the "func" keyword. We can assume that the nearest "@inline" before this offset is
+        // the attribute we are interested in.
+        guard functionKinds.contains(kind),
+            let array = dictionary["key.attributes"] as? [Any],
+            case let attributes = array.flatMap({ ($0 as? [String: String])?["key.attribute"] }),
```

Comment:
There's a private [`extractAttributes(_:)`](https://github.com/realm/SwiftLint/blob/0.13.2/Source/SwiftLintFramework/Rules/OverriddenSuperCallRule.swift#L92-L98) function elsewhere that might be worth putting in a shared location and reusing, since we extract attributes like this in a few rules already (`OverriddenSuperCallRule`, `PrivateOutletRule`, `ValidIBInspectableRule` and `WeakDelegateRule`).